core/src/apps/bitcoin/sign_tx/bitcoin.py

```diff
@@ -153,9 +154,12 @@ async def step3_confirm_tx(self) -> None:
 
         total = self.total_in - self.change_out
         spending = total - self.external_in
+        # fee_threshold = (coin.maxfee per byte * tx size)
+        fee_threshold = (self.coin.maxfee_kb / 1000) * (self.weight.get_total() / 4)
 
-        # fee > (coin.maxfee per byte * tx size)
-        if fee > (self.coin.maxfee_kb / 1000) * (self.weight.get_total() / 4):
+        if fee > fee_threshold:
+            if fee > 10 * fee_threshold and not device.unsafe_prompts_allowed():
+                raise wire.DataError("Fee exceeds hardcoded limit")
```

Comment:
The error seems a bit confusing. How about "The fee is unexpectedly large"?

Reply:
Fixed 426f9511daff5023c186d12ea89ea2823d5526c5.